torch/export/_unlift.py

```diff
@@ -73,20 +81,106 @@ def _check_inputs_match(args, kwargs, in_spec: pytree.TreeSpec) -> list:
     return flat_args_with_path
 
 
-@torch._dynamo.disable
-def _check_input_constraints_pre_hook(self, args, kwargs):
-    if not self.validate_inputs:
-        return
+def _convert_guards_code_to_fn(
+    guards_code: list[str],
+    paths_of_placeholders: list[pytree.KeyPath],
+):
+    """
+    Generates Python code given guards code and paths of placeholders.
+    We assume that, based on source information,
+    - the tracer generates the guards code
+    - the input spec generates the paths of placeholders.
+
+    Example:
+
+    Suppose we are given the guards code "L['z']['k'].size()[1] == 3"
+    and we are given that ['z']['k'] is the path of placeholder #2.
+    Then we will generate:
+    ```
+    torch._assert(
+        args[2].size()[0] == 3,
+        "Guard failed: z['k'].size()[0] == 3",
+    )
+    ```
+
+    FAQ: Why do we generate code based on (flattened) args instead of
```

Comment:
Maybe i should indeed just pass in shape env with flat_args directly here? So you dont need to manually flatten inputs again?

Reply:
I need to flatten only to get the keys (i.e., paths to placeholders). If the keys were made available some other way maybe it would work too.